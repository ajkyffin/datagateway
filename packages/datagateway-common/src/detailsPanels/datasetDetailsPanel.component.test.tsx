import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import { Dataset } from '../app.types';
import DatasetDetailsPanel from './datasetDetailsPanel.component';

describe('Dataset details panel component', () => {
  let shallow;
  let rowData: Dataset;
  const detailsPanelResize = jest.fn();

  beforeEach(() => {
    shallow = createShallow();
    rowData = [
      {
        id: 1,
        name: 'Test 1',
        size: 1,
        modTime: '2019-07-23',
        createTime: '2019-07-23',
      },
    ];
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const wrapper = shallow(
      <DatasetDetailsPanel
        rowData={rowData}
        detailsPanelResize={detailsPanelResize}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
