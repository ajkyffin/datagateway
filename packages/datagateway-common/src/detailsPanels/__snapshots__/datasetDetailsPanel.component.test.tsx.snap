// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dataset details panel component renders correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-root-1"
  container={true}
  direction="column"
  id="details-panel"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      variant="h6"
    >
      <b />
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Divider))
      className="makeStyles-divider-2"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      variant="overline"
    >
      datasets.description
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))>
      <b />
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
